components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/listener/IdentityClaimValueEncryptionListener.java

```diff
@@ -190,12 +200,18 @@ private void updateClaimValues(Map<String, String> claims, UserStoreManager user
      * @param plainText text to be encrypted.
      * @return encrypted claim value.
      */
-    private String encryptClaimValue(String plainText) throws CryptoException {
+    private String encryptClaimValue(String plainText, boolean isCustomKeyEnabled) throws CryptoException {
 
         if (plainText.isEmpty()) {
             return plainText;
         }
-        return CryptoUtil.getDefaultCryptoUtil().encryptAndBase64Encode(plainText.getBytes(StandardCharsets.UTF_8));
+        // Get custom key from server configuration.
+        String customKey = null;
+        if (isCustomKeyEnabled) {
+            customKey = ServerConfiguration.getInstance().getFirstProperty(TOTP_KEY);
+        }
+        return CryptoUtil.getDefaultCryptoUtil().encryptAndBase64Encode(
```

Comment:
If the customKey is null, will it be handle by `CryptoUtil.getDefaultCryptoUtil().encryptAndBase64Encode` method?

Reply:
Yes, it will be handle in this method: https://github.com/wso2/carbon-kernel/blame/4.9.x/core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/CryptoUtil.java#L136